### ISSUE TITLE
[Bug] fit_gpytorch_model takes long time to fit the model

### ISSUE THREAD

**Issue by roshni-kamath** [pytorch/botorch#456]
# 🐛 Bug
I am trying to fit a GP model, but it takes more than 3 hours to fit the model on CPU, can you please give me some insights into why it takes so much time?

## To reproduce

** Code snippet to reproduce **
```python
def initialize_model(train_x, train_obj, state_dict=None):
        train_x = normalize(train_x, bounds)
        train_obj = standardize(train_obj)

        model_obj = SingleTaskGP(train_X=train_x, train_Y=train_obj)
        mll = ExactMarginalLogLikelihood(model_obj.likelihood, model_obj)
        # load state dict if it is passed
        if state_dict is not None:
            model_obj.load_state_dict(state_dict)

        return mll, model_obj


train_x, train_obj = generate_initial_data() // This generates around 12000 rows with 11 columns
mll, model = initialize_model(train_x, train_obj)
# fit the models
fit_gpytorch_model(mll)
```

## System information

**Please complete the following information:**
-  BoTorch Version -- 0.2.2
- GPyTorch Version --> 1.0.1
- PyTorch Version  --> 1.3.1
- Computer OS --> Ubuntu

## Additional context

I noticed that when the data is normalized, standardized and then the GP is fit_gpytorch_model, it takes more than when the data is not normalized, standardized.

**Comment by sdaulton** [comment]
Hi Roshni,

By default `fit_gpytorch_model` computes the marginal log likelihood (MLL) exactly, which scales cubically with the number of training examples. It would take a very long time with 12,000 training examples. One option is to approximate the MLL using linear CG via `fit_gpytorch_model(mll, approx_mll=True)`. In addition, you'll see a massive speedup if you run this on a GPU. 

```
from botorch.models import SingleTaskGP
import torch
from botorch.fit import fit_gpytorch_model
from gpytorch.mlls import ExactMarginalLogLikelihood
from botorch.utils import standardize

def initialize_model(train_x, train_obj, state_dict=None):
#         train_x = normalize(train_x, bounds)
        train_obj = standardize(train_obj)

        model_obj = SingleTaskGP(train_X=train_x, train_Y=train_obj)
        mll = ExactMarginalLogLikelihood(model_obj.likelihood, model_obj)
        # load state dict if it is passed
        if state_dict is not None:
            model_obj.load_state_dict(state_dict)

        return mll, model_obj

train_x = torch.rand(12000,11, dtype=torch.double, device=torch.device("cuda:4"))
train_obj = (torch.sin(train_x) + torch.cos(10*train_x)+ torch.randn_like(train_x[0])).sum(dim=-1, keepdim=True)


def fit_model():
    mll, model = initialize_model(train_x, train_obj)
    # fit the models
    fit_gpytorch_model(mll, approx_mll=True)

%timeit fit_model()
```

Output:
```9.05 s ± 2.59 s per loop (mean ± std. dev. of 7 runs, 1 loop each)```


> I noticed that when the data is normalized, standardized and then the GP is fit_gpytorch_model, it takes more than when the data is not normalized, standardized.

The default priors used in `SingleTaskGP` assume that the inputs are normalized (`[0,1]^d`) and the outputs are standardized (`N(0,1)`). If the inputs and outputs are not normalized and standardized respectively, the priors are misspecified and model fitting will likely terminate quickly with a suboptimal model fit.

**Comment by roshni-kamath** [comment]
@sdaulton can you please cite some paper or reference so I can look into this issue further?

**Comment by sdaulton** [comment]
@roshni-kamath see https://papers.nips.cc/paper/7985-gpytorch-blackbox-matrix-matrix-gaussian-process-inference-with-gpu-acceleration.pdf for more on linear CG in gpytorch
